### ISSUE TITLE
[Shell Panel] Emit "Resize End" Event

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-components#4037]
### Description

It would be nice to have a "resize end" event emit the pixel width of the component when a user finishes interaction. For mouse users emitting on drag end, for keyboard users it could emit on each adjustment.

### Acceptance Criteria

Add an event that emits the pixel width of the shell panel component after resize.

### Which Component

Shell Panel

### Example Use Case

As a dev or designer, after a user resizes the shell panel component, I'd like to save that px value in localStorage.

**Comment by driskull** [comment]
@macandcheese I thought about this a while back but shouldn't a user be able to setup a resize observer in this case?

There are a lot of components that may change size depending on interaction and I don't think we want to try and add events for all of them.

**Comment by macandcheese** [comment]
> There are a lot of components that may change size depending on interaction and I don't think we want to try and add events for all of them.

Yes agreed it wouldn't make sense generally - this one seems it could be expected / justified as the resize functionality is offered by the component.

That said, it's a nice-to-have, folks (read: me 😆) can set it up on their own as you mention.

**Comment by jcfranco** [comment]
I agree 100% with @driskull. Closing.
